### DISCUSSION TITLE
Impossible to sync git repository with Azure DevOps since 2.5.272

### DISCUSSION THREAD

**Discussion by laellyo** [requarks/wiki#6045] [Error / Bug Report]
We have an issue since few weeks with the GIT synchronization capability of WikiJS.

### Actual behavior
Without any change on Azure DevOps credentials (SSH or PAT), we meet an error when we deploy the WikiJS version 2.5.272 or above. 

For SSH:

> Unable to negotiate with XX.XX.XX.XX port 22: no matching host key type found. Their offer: ssh-rsa fatal: Could not read from remote repository. Please make sure you have the correct access rights and the repository exists. error: Could not fetch origin 

For PAT:
> fatal: Authentication failed for 'https://dev.azure.com/xxxxx/yyyyyy/_git/wiki/' error: Could not fetch origin 

In logs, the only related logs I found are:

> 2022-02-28T15:51:44.837622999Z 2022-02-28T15:51:44.837Z [MASTER] [33mwarn[39m: fatal: ambiguous argument 'staging': unknown revision or path not in the working tree.
> 2022-02-28T15:51:44.837872402Z Use '--' to separate paths from revisions, like this:
> 2022-02-28T15:51:44.837883702Z 'git <command> [<revision>...] -- [<file>...]'

### Expected behavior

If we rollback to version 2.5.268, everything works fine again, without changing any GIT settings in WikiJS.

### Steps to reproduce the behavior

If a dedicated Azure DevOps repo + PAT is required to help you to find the root cause, feel free to contact me. I will do my best to provide you temporary resources.

Thanks in advance for your help !

**Comment by maffelbaffel** [comment]
Same issue unfortunatly. We are using the currently latest version 2.5.277 
Edit: This is not limited to Azure DevOps. We get the same error with Github.

**Comment by adketuri** [comment]
I ran into the same issue following the setup instructions outlined in the [documentation](https://docs.requarks.io/storage/git), using the DigitalOcean one-click deploy. I noticed running `ssh -T git@github.com` from my droplet connected to github successfully, but not within the docker container (ie `docker container exec -it wiki ssh -T git@github.com` would give me the publickey error). 

My workaround was to generate the keys within the container:

`docker container exec -it wiki ssh-keygen -t rsa -b 4096`

And then just dropping the private/public keys in the respective admin panel location/github deploy keys page, outputting the key contents with:

`docker container exec -it wiki cat /home/node/.ssh/id_rsa.pub`

This is not a great solution -- I'm not a docker expert but I believe these will just get nuked with a rebuild -- but for now at least it seems syncing works. Curious if anyone has a better solution, but hope this is helpful.
